internal/controllers/topology/cluster/cluster_controller.go

```diff
@@ -228,7 +229,11 @@ func (r *Reconciler) reconcile(ctx context.Context, s *scope.Scope) (ctrl.Result
 	// Note: This doesn't require requeue as a change to ClusterClass observedGeneration will cause an additional reconcile
 	// in the Cluster.
 	if clusterClass.GetGeneration() != clusterClass.Status.ObservedGeneration {
-		return ctrl.Result{}, nil
+		return ctrl.Result{}, errors.Errorf("ClusterClass is not successfully reconciled: ClusterClass.status.observedGeneration must be %d, but is %d", clusterClass.GetGeneration(), clusterClass.Status.ObservedGeneration)
+	}
+	if !conditions.Has(clusterClass, clusterv1.ClusterClassVariablesReconciledCondition) ||
```

Comment:
Realized that the Cluster topology controller was not block in this case.

Previous behavior was enough for inline variables, because with every change of inline variables generation was changed as well.

But if now variables are coming from a RuntimeExtension just checking the generation is not enough anymore

Reply:
Nice catch!